packages/truffle-db/src/workspace/pouch.ts

```diff
@@ -162,7 +172,94 @@ export class Workspace {
     };
   }
 
+  async contractInstance ({ id }: { id: string }) {
+    await this.ready;
+
+    try {
+      return {
+        ...await this.contractInstances.get(id),
+
+        id
+      };
+    } catch (_) {
+      return null;
+    }
+  }
+
+  async contractInstancesAdd ({ input }) {
+    await this.ready;
+
+    const { contractInstances } = input;
+
+    return {
+      contractInstances: Promise.all(contractInstances.map(
+        async (contractInstanceInput) => {
+          const { address, network, contract, callBytecode } = contractInstanceInput;
+          // hash includes address and network of this contractInstance
+          const id = soliditySha3(jsonStableStringify({
+            address: address,
+            network: network
```

Comment:
Since we only want the `network.id` portion of this to be used, it's probably worth being explicit about this here:

```suggestion
            network: { id: network.id }
```